search/src/Components/upload/react-images-upload/index.js

```diff
@@ -87,53 +93,38 @@ class ReactImageUploadComponent extends React.Component {
         fileErrors.push(fileError);
         continue;
       }
-
-      allFilePromises.push(this.readFile(file));
-    }
-
-    this.setState({
-      fileErrors
-    });
-
-    const {singleImage} = this.props;
-
-    Promise.all(allFilePromises).then(newFilesData => {
-      const dataURLs = singleImage?[]:this.state.pictures.slice();
-      const files = singleImage?[]:this.state.files.slice();
-
-      newFilesData.forEach(newFileData => {
-        this.uploadFileToServer(dataURLs, files, newFileData);
+      this.readFile(file).then(newFileData => {
+        this.uploadFileToServer(newFileData);
       });
-    });
+    }
+    this.setState({fileErrors});
   }
 
   /*
      Customised function modified on original codebase
      Copyright (c) 2020 Zheng Li
    */
-  uploadFileToServer(dataURLs, files, newFileData) {
-    const filesName = files.map(file => file.name);
-    if (!this.props.images.includes(newFileData.file.name) || filesName.includes(newFileData.file.name)){
-        return
-    }
-    else {
-        const body = new FormData()
-        body.append('upload_image', newFileData.file)
-        body.append('upload_id', this.props.upload_id)
-        axios({
-            method: 'post',
-            url: this.props.uploadURL,
-            data: body,
-            headers: {'Content-Type': 'multipart/form-data', 'X-CSRFToken': Cookies.get('csrftoken') }
-        }).then(res => {
-            if(res.status === 200){
-                dataURLs.push(newFileData.dataURL);
-                files.push(newFileData.file);
-                this.setState({pictures: dataURLs, files: files});
-                this.props.handleUploaded(files.map(file => file.name));
-            }
-        })
-    }
+  uploadFileToServer(newFileData) {
+      const {singleImage} = this.props;
+      const body = new FormData()
+      body.append('upload_image', newFileData.file)
+      body.append('upload_id', this.props.upload_id)
+      axios({
+          method: 'post',
+          url: this.props.uploadURL,
+          data: body,
+          headers: {'Content-Type': 'multipart/form-data', 'X-CSRFToken': Cookies.get('csrftoken') }
+      }).then(res => {
+          if(res.status === 200){
+            const dataURLs = singleImage?[]:this.state.pictures.slice();
+            const files = singleImage?[]:this.state.files.slice();
+            dataURLs.push(newFileData.dataURL);
+            files.push(newFileData.file);
+            this.setState({pictures: dataURLs, files: files});
+            this.props.handleUploaded(files.map(file => file.name));
```

Comment:
This needs to ensure uniqueness, or syncImageErrorMessage needs to ensure uniqueness. Ideally syncImageErrorMessage (and maybe the setting of `pictures` and `files` in state) would be debounced to avoid updating too rapidly. We can discuss.